src/Executor/Runner/Adapter.php

```diff
@@ -111,9 +131,25 @@ abstract public function getRuntime(string $name): mixed;
 
     abstract public function getStats(): Stats;
 
-    protected function getStorageDevice(string $root): Device
-    {
-        $connection = Http::getEnv('OPR_EXECUTOR_CONNECTION_STORAGE', '');
+    protected function getStorageDevice(
+        string $root,
+        string $region = ''
+    ): Device {
+        $connections = System::getEnv('OPR_EXECUTOR_CONNECTION_STORAGE', '') ?? '';
+
+        if (\preg_match('/^\w+=/', $connections)) {
```

Comment:
what is this check? Cant we explode by comma, and have array of connections? then while looping, code would only run once?

Reply:
Checks if it starts with any word then equals sign. We can't do loop and split, because we have to support old style for self-hosted, and doing split on equals there breaks at the query params because there is not `region=`

This way, we ensure the we have `something=` at the very start of the string